### PR TITLE
[FX-5057] Migrate ButtonAction to TailwindCSS

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-5057]

### Description

We are going to migrate ButtonAction to TailwindCSS and new MUI Base Button

### How to test

- [Temploy](https://picasso.toptal.net/fx-5057-migrate-button-action)

### Development checks

- N/A Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- N/A Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5057]: https://toptal-core.atlassian.net/browse/FX-5057?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/base/Button/src/ButtonBase/ButtonBase.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import type {
```
Comment (dmaklygin) [inline packages/base/Button/src/ButtonBase/ButtonBase.tsx:1]: I created a separate ButtonBase component that will be reused by other Buttons

File: packages/picasso-tailwind/src/index.js [hunk 2]
```diff
@@ -98,8 +98,6 @@ module.exports = {
       22: '0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)',
       23: '0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)',
       24: '0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)',
-      button: '0 0 0 3px rgba(32, 78, 207, 0.48)',
-      'button-transparent': '0 0 0 3px rgba(255, 255, 255, 0.48)',
```
Comment (dmaklygin) [inline packages/picasso-tailwind/src/index.js:102]: Removed as we decided before

File: packages/base/Button/src/ButtonBase/styles.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  if (!disabled && !loading) {
+    classNames.push('cursor-pointer')
+  } else {
+    classNames.push('cursor-default')
+    classNames.push('pointer-events-none')
+  }
+
+  return classNames
+}
```
Comment (dmaklygin) [inline packages/base/Button/src/ButtonBase/styles.ts:37]: ButtonBase contains only necessary, core styles that are related to all button types

**Review by TomasSlama — APPROVED** [review]
Looks good! I only noticed that we used to have `height: 1.5em` set before, now we don't

**Comment by oliviertassinari** [comment]
Cool to see Toptal on https://twitter.com/Base_UI

**Pull request merged**